WebCore::XPath::Value has uninitialized fields after construction

ddkilzer opened the pull request:
#### c4467fc5aed587478747535b46aedf0d08ad2995
<pre>
WebCore::XPath::Value has uninitialized fields after construction
&lt;<a href="https://bugs.webkit.org/show_bug.cgi?id=274864">https://bugs.webkit.org/show_bug.cgi?id=274864</a>&gt;
&lt;<a href="https://rdar.apple.com/128966823">rdar://128966823</a>&gt;

Reviewed by Chris Dumez.

Fix in XPathValue.h is to set m_bool and m_double to default values when
constructing WebCore::XPath::Value objects.

Additional drive-by fixes that comprise most of the changes:
- Update WebCore::XPath::Value::Type to an enum class.
- Fix format of header files to remove unwanted whitespace.
- Add missing comments for closing namespace braces.
- Fix indentation of case statements in XPathValue.cpp.
- Fix formatting of constructors in XPathValue.h.

* Source/WebCore/xml/XPathFunctions.cpp:
* Source/WebCore/xml/XPathPath.h:
* Source/WebCore/xml/XPathPredicate.cpp:
(WebCore::XPath::predicateIsContextPositionSensitive):
* Source/WebCore/xml/XPathPredicate.h:
* Source/WebCore/xml/XPathResult.cpp:
(WebCore::XPathResult::XPathResult):
* Source/WebCore/xml/XPathValue.cpp:
(WebCore::XPath::Value::modifiableNodeSet):
(WebCore::XPath::Value::toBoolean const):
(WebCore::XPath::Value::toNumber const):
(WebCore::XPath::Value::toString const):
* Source/WebCore/xml/XPathValue.h:
(WebCore::XPath::Value):
- Initialize m_bool to false and m_double to 0 to fix the bug.
(WebCore::XPath::Value::Type):
- Change to enum class with size uint8_t.
- Remove redundant &quot;Value&quot; from the end of enum names.
(WebCore::XPath::Value::Value):
- Delete default constructor since it is unused.
(WebCore::XPath::Value::type const):
(WebCore::XPath::Value::isNodeSet const):
(WebCore::XPath::Value::isBoolean const):
(WebCore::XPath::Value::isNumber const):
(WebCore::XPath::Value::isString const):
(WebCore::XPath::Value::Data::create):
(WebCore::XPath::Value::Data::Data):
* Source/WebCore/xml/XPathVariableReference.h:

Canonical link: <a href="https://commits.webkit.org/279506@main">https://commits.webkit.org/279506@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e4ba0abd95dc9c7e56e9a473e4a2ba857f1dc312

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/53632 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/32992 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/6144 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/56912 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/4358 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/55935 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/40472 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/4171 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/43450 "Found 19 new test failures: webgl/2.0.0/conformance/canvas/rapid-resizing.html, webgl/2.0.y/conformance2/textures/canvas/tex-3d-r11f_g11f_b10f-rgb-float.html, webgl/2.0.y/conformance2/textures/canvas/tex-3d-r11f_g11f_b10f-rgb-half_float.html, webgl/2.0.y/conformance2/textures/canvas/tex-3d-r8-red-unsigned_byte.html, webgl/2.0.y/conformance2/textures/canvas/tex-3d-rgb10_a2-rgba-unsigned_int_2_10_10_10_rev.html, webgl/2.0.y/conformance2/textures/canvas/tex-3d-rgb9_e5-rgb-half_float.html, webgl/2.0.y/conformance2/textures/canvas/tex-3d-rgba16f-rgba-half_float.html, webgl/2.0.y/conformance2/textures/canvas/tex-3d-rgba32f-rgba-float.html, webgl/2.0.y/conformance2/textures/canvas/tex-3d-rgba8-rgba-unsigned_byte.html, webgl/2.0.y/conformance2/textures/canvas/tex-3d-srgb8-rgb-unsigned_byte.html ... (failure)") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/2843 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/55729 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/31212 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/46366 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/24585 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/28038 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/3683 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/2513 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/49774 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/3864 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/58507 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/28794 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/3888 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/50867 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/29994 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/46528 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/50207 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/30925 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/7917 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/29771 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/xml/XPathValue.h:
```diff
@@ -29,75 +29,87 @@
+
+class Value {
+public:
+    enum class Type : uint8_t { NodeSetValue, BooleanValue, NumberValue, StringValue };
```
nit: Not sure we really need the "Value" suffix for each value 🤷🏻

ddkilzer replied:
Ah, good call.  That was the previous name.  I can fix this before landing.

Commit:
WebCore::XPath::Value has uninitialized fields after construction
<https://bugs.webkit.org/show_bug.cgi?id=274864>
<rdar://128966823>

Reviewed by Chris Dumez.

Fix in XPathValue.h is to set m_bool and m_double to default values when
constructing WebCore::XPath::Value objects.

Additional drive-by fixes that comprise most of the changes:
- Update WebCore::XPath::Value::Type to an enum class.
- Fix format of header files to remove unwanted whitespace.
- Add missing comments for closing namespace braces.
- Fix indentation of case statements in XPathValue.cpp.
- Fix formatting of constructors in XPathValue.h.

* Source/WebCore/xml/XPathFunctions.cpp:
* Source/WebCore/xml/XPathPath.h:
* Source/WebCore/xml/XPathPredicate.cpp:
(WebCore::XPath::predicateIsContextPositionSensitive):
* Source/WebCore/xml/XPathPredicate.h:
* Source/WebCore/xml/XPathResult.cpp:
(WebCore::XPathResult::XPathResult):
* Source/WebCore/xml/XPathValue.cpp:
(WebCore::XPath::Value::modifiableNodeSet):
(WebCore::XPath::Value::toBoolean const):
(WebCore::XPath::Value::toNumber const):
(WebCore::XPath::Value::toString const):
* Source/WebCore/xml/XPathValue.h:
(WebCore::XPath::Value):
- Initialize m_bool to false and m_double to 0 to fix the bug.
(WebCore::XPath::Value::Type):
- Change to enum class with size uint8_t.
- Remove redundant "Value" from the end of enum names.
(WebCore::XPath::Value::Value):
- Delete default constructor since it is unused.
(WebCore::XPath::Value::type const):
(WebCore::XPath::Value::isNodeSet const):
(WebCore::XPath::Value::isBoolean const):
(WebCore::XPath::Value::isNumber const):
(WebCore::XPath::Value::isString const):
(WebCore::XPath::Value::Data::create):
(WebCore::XPath::Value::Data::Data):
* Source/WebCore/xml/XPathVariableReference.h:

Canonical link: https://commits.webkit.org/279506@main
